src/features/filter.ts

```diff
@@ -1,12 +1,28 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
 
-const initialState = {
+type Filter = {
+  query: string;
+  status: Status;
+};
+
+const initialState: Filter = {
   query: '',
-  status: 'all',
+  status: Status.all,
```

Comment:
The `Status.all` might be incorrect if `Status` is an enum and does not have an `all` value. Ensure that `Status.all` is a valid value in the `Status` enum.